pkg/utils/test/events.go

```diff
@@ -0,0 +1,42 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package test
+
+import (
+	"testing"
+	"time"
+
+	"k8s.io/client-go/tools/record"
+)
+
+const (
+	// EventsTimeout is na arbitrary timeout for events to be collected in a unit test.
+	EventsTimeout = time.Second * 5
+)
+
+// ReadAtMostEvents attempts to read at most minEventCount from a FakeRecorder.
+func ReadAtMostEvents(t *testing.T, minEventCount int, recorder *record.FakeRecorder) []string {
+	t.Helper()
+	if minEventCount == 0 {
+		return nil
+	}
+	c := time.After(EventsTimeout)
+	gotEvents := make([]string, 0, minEventCount)
+	gotEventCount := 0
+	for {
+		select {
+		case e := <-recorder.Events:
+			gotEvents = append(gotEvents, e)
+			gotEventCount++
+			if gotEventCount == minEventCount {
+				// We have read the min. expected number of events
+				return gotEvents
+			}
+		case <-c:
```

Comment:
Do we need the timeout given that this is always used in a unit test context and there is no expectation of concurrent use? 

```golang
	t.Helper()
	if minEventCount == 0 {
		return nil
	}
	gotEvents := make([]string, 0, minEventCount)
	gotEventCount := 0
	for i := 0; i < minEventCount; i++ {
		select {
		case e := <-recorder.Events:
			gotEvents = append(gotEvents, e)
			gotEventCount++
		default:
			t.Errorf("Timeout: expected at least %d events, got %d", minEventCount, gotEventCount)
		}
	}
	return gotEvents
```

also the name is could be then something along the lines of  `ReadNEvents`

Reply:
> Do we need the timeout given that this is always used in a unit test context and there is no expectation of concurrent use?

I think we need a timeout so we eventually exit the function if there are less events than `minEventCount`. Otherwise, the function may wait forever on `case e := <-recorder.Events`.

Reply:
That's why I added the default clause in the `select` so that it immediately errors out or maybe I am missing your point.

Reply:
Sorry I'm missed your original point. I was assuming that we didn't want to depend on whether or not events were published synchronously in the recorder. But I agree that it is fair to say that it will be the case here.